From 8951559e004fd7470b99eb9f8eaf3a455ef5bbee Mon Sep 17 00:00:00 2001
From: Trevor Tao <trevor.tao@arm.com>
Date: Fri, 12 Nov 2021 00:14:54 +0800
Subject: Enable building for base images (#35994)

Now for DOCKER_V2_BUILDER, it does not support the explicit building for
base images, for example, when building with
make dockerx.app_sidecar_base_ubuntu_xenial
it would show some error information for the wrong VM_IMAGE_NAME and
VM_IMAGE_VERSION, now we just fix it here.

And add other base images in the istio-docker.mk too.

Signed-off-by: trevor.tao <trevor.tao@arm.com>
---
 tools/docker-builder/main.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/tools/docker-builder/main.go b/tools/docker-builder/main.go
index 1cac95a3f3..24c8c889cf 100644
--- a/tools/docker-builder/main.go
+++ b/tools/docker-builder/main.go
@@ -287,6 +287,10 @@ func vmImageName(target string) string {
 		// Not a VM
 		return ""
 	}
+	if strings.HasPrefix(target, "app_sidecar_base") {
+		return strings.Split(target, "_")[3]
+	}
+
 	return strings.Split(target, "_")[2]
 }
 
@@ -295,6 +299,10 @@ func vmImageVersion(target string) string {
 		// Not a VM
 		return ""
 	}
+	if strings.HasPrefix(target, "app_sidecar_base") {
+		return strings.Split(target, "_")[4]
+	}
+
 	return strings.Split(target, "_")[3]
 }
 
-- 
2.35.3

